### PR TITLE
chore: create CHANGELOG file

### LINKED ISSUE #118: Create CHANGELOG file content for welcome dialog
This is the file that is linked by #102.

https://github.com/zd-zero/waifu-motivator-plugin/blob/bde25b786129bec54a4f1d18ab7e2bfb63113b4b/src/main/java/zd/zero/waifu/motivator/plugin/onboarding/UpdateNotification.kt#L18

### PR DESCRIPTION
**Pull request by zd-zero**
Resolves #118 

Created a `CHANGELOG.md` file for the project and moved the `/parts` into `/docs` and converted them to markdown files which would be converted back to HTML during the build process.

Built the plugin and can properly see the converted markdown files to HTML files.
Screen Shot 2020-08-06 at 21 16 46

### CODE REVIEW

File: parts/pluginChanges.html [hunk 1]
```diff
@@ -4,82 +4,4 @@
...
-<p>v0.1.1</p>
-<ul>
-    <li>Bumped IDEA for compatibility to >= 2019.3</li>
-</ul>
-
-<p>v0.1</p>
-<ul>
-    <li>Initial EAP Release for IDEA >= 2019.2.3</li>
-</ul>
+<a href="https://github.com/zd-zero/waifu-motivator-plugin/blob/master/CHANGELOG.md">View Changelog</a>
```
Comment (zd-zero) [inline parts/pluginChanges.html:7]: I'm thinking if we can remove this file and use the CHANGELOG to the gradle change file
https://github.com/zd-zero/waifu-motivator-plugin/blob/bde25b786129bec54a4f1d18ab7e2bfb63113b4b/gradle.properties#L9

for it was used in build.gradle
https://github.com/zd-zero/waifu-motivator-plugin/blob/bde25b786129bec54a4f1d18ab7e2bfb63113b4b/build.gradle#L57

I don't know if Jetbrains Marketplace will properly read the markdown file for the page.

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
+# Changelog
```
Comment (zd-zero) [inline CHANGELOG.md:1]: Opt to manually create changelog file content, then I'll check if we can auto-generate the changelog for versions as we're already using [conventional commit message](https://www.conventionalcommits.org/en/v1.0.0/).

File: parts/pluginChanges.html (hunk 1 above)
Comment (Unthrottled) [reply parts/pluginChanges.html:7]: So what we can do is just keep the markdown and then convert it to HTML during the build process.

I've done that before. You can take a look at this PR if you want to https://github.com/one-dark/jetbrains-one-dark-theme/pull/157/files

Comment (Unthrottled) [reply parts/pluginChanges.html:7]: > use the CHANGELOG

Personally, I would keep the change notes concise to the new things in the latest version. Then keep the link to the changelog if they want to know all the changes.

I have not come up with anything better other than keeping a Changelog and a [release notes](https://github.com/doki-theme/doki-theme-jetbrains/blob/master/changelog/RELEASE-NOTES.md).

**Comment by zd-zero** [comment]
For now, I'll try to convert `CHANGELOG.md` into `build.gradle`'s `changeNotes`. It looks like it'll be simple but I'm struggling to do it 😅.

**Comment by Unthrottled** [comment]
You'll want to convert your markdown to html.

Here is the gradle plugin I use: https://github.com/kordamp/markdown-gradle-plugin

Here is `build.gradle` with the markdownToHtml configured: https://github.com/doki-theme/doki-theme-jetbrains/blob/master/build.gradle

Let me know if you need anything more :+1:

File: build.gradle [hunk 3]
```diff
@@ -49,10 +48,25 @@ compileTestKotlin {
     kotlinOptions.jvmTarget = "1.8"
 }
 
+
+markdownToHtml {
+    sourceDir file("$projectDir/docs")
```
Comment (zd-zero) [inline build.gradle:53]: Turns out that I can move the `CHANGELOG.md` file into `docs` dir with the description file as I was trying to create a script to copy the `CHANGELOG.md` file into a dir and use it as a source, but found out that I can also turn the description altogether which will result in a simpler solution.

File: src/main/java/zd/zero/waifu/motivator/plugin/onboarding/UpdateNotification.kt [hunk 4]
```diff
@@ -15,7 +15,7 @@ val UPDATE_MESSAGE: String = """
         <li>Visual Test Pass Notifications!</li>
         <li>More settings customization.</li>
       </ul>
-      <br>Please see the <a href="https://github.com/zd-zero/waifu-motivator-plugin/blob/master/CHANGELOG.md">changelog</a> for more details.
+      <br>Please see the <a href="https://github.com/zd-zero/waifu-motivator-plugin/blob/master/docs/CHANGELOG.md">changelog</a> for more details.
```
Comment (zd-zero) [inline src/main/java/zd/zero/waifu/motivator/plugin/onboarding/UpdateNotification.kt:18]: Updated the path; `CHANGELOG.md` was now moved to `/docs` dir.

**Comment by Unthrottled** [comment]
This looks good!

Once we implement a CI deployment we need to remember to remember to follow these steps:

```bash
# Generate the HTML so that the `patchPluginXml` task's `filExists` passes in the next step
./gradlew markdowntohtml

# build the plugin because the `html` is present in the `build` directory
./gradlew buildPlugin
```

Because if you run a 
`./gradlew clean`
(which simulates a freshly cloned repo). Then run
`./gradlew patchPluginXml` 
The `build/patchedPluginXmlFiles` contains an xml file with no change notes :disappointed:

**Comment by zd-zero** [comment]
> remember to follow these steps

Is there a way to prevent this? Or just really add a reminder?

**Comment by Unthrottled** [comment]
Not that I am aware of, I had to do the same thing.
https://github.com/doki-theme/doki-theme-jetbrains/blob/1464f9ec6853471aa57cce2604ccc55ea562b4a6/ciScripts/buildPlugin.sh#L9

You can merge it if you are good.

**Pull request merged**